tests/testthat/test-step-call-pivot_wider.R

```diff
@@ -0,0 +1,210 @@
+test_that("can pivot all cols to wide", {
+  df <- lazy_dt(tibble(key = c("x", "y", "z"), val = 1:3), "DT")
+  step <- pivot_wider(df, names_from = key, values_from = val)
+  pv <- collect(step)
+
+  expect_equal(step$vars, c("x", "y", "z"))
+  expect_equal(nrow(pv), 1)
+  expect_equal(
+    show_query(step),
+    expr(dcast(DT, formula = "... ~ key", value.var = "val")[, .(x, y, z)])
+  )
+})
+
+test_that("non-pivoted cols are preserved", {
+  df <- lazy_dt(tibble(a = 1, key = c("x", "y"), val = 1:2), "DT")
+  step <- pivot_wider(df, names_from = key, values_from = val)
+  pv <- collect(step)
+
+  expect_equal(step$vars, c("a", "x", "y"))
+  expect_equal(nrow(pv), 1)
+  expect_equal(
+    show_query(step),
+    expr(dcast(DT, formula = "a ~ key", value.var = "val"))
+  )
+})
+
+test_that("implicit missings turn into explicit missings", {
+  df <- lazy_dt(tibble(a = 1:2, key = c("x", "y"), val = 1:2))
+  pv <- collect(pivot_wider(df, names_from = key, values_from = val))
+
+  expect_equal(pv$a, c(1, 2))
+  expect_equal(pv$x, c(1, NA))
+  expect_equal(pv$y, c(NA, 2))
+})
+
+test_that("error when overwriting existing column", {
+  df <- tibble(
+    a = c(1, 1),
+    key = c("a", "b"),
+    val = c(1, 2)
+  )
+  df <- lazy_dt(df)
+  expect_error(
+    pivot_wider(df, names_from = key, values_from = val),
+    "Names must be unique"
+  )
+})
+
+test_that("grouping is preserved", {
+  df <- lazy_dt(tibble(g = 1, k = "x", v = 2))
+  out <- df %>%
+    group_by(g) %>%
+    pivot_wider(names_from = k, values_from = v)
+  expect_equal(out$groups, "g")
+})
+
+# https://github.com/tidyverse/tidyr/issues/804
+test_that("column with `...j` name can be used as `names_from`", {
+  df <- lazy_dt(tibble(...8 = c("x", "y", "z"), val = 1:3))
+  step <- pivot_wider(df, names_from = ...8, values_from = val)
+  pv <- collect(step)
+  expect_equal(step$vars, c("x", "y", "z"))
+  expect_equal(nrow(pv), 1)
+})
+
+# column names -------------------------------------------------------------
+
+test_that("names_glue affects output names & auto-converts data.table to lazy_dt", {
+  df <- data.table(
+    x = c("X", "Y"),
+    y = 1:2,
+    a = 1:2,
+    b = 1:2
+  )
+
+  step <- pivot_wider(df, names_from = x:y, values_from = a:b, names_glue = "{x}{y}_{.value}")
```

Comment:
Here I'd suggest an `expect_snapshot(show_query(step))`, just to capture the sequence of calls that you're generating.  Similarly for the test for `names_sort = TRUE`